### PR TITLE
[openhabcloud] Upgrade dependencies, use DS annotations, add license info

### PR DESCRIPTION
**Pull request by wborn**
I've upgraded the openHAB Cloud Connector dependencies to see if it solves the reconnection issues where you need to restart the bundle to get your instance online again. 

So far the upgrades don't seem to introduce new issues. Maybe it also helps with solving issues like #1700, #3031. The OkHttp library has a lot of connection related [fixes](https://github.com/square/okhttp/blob/master/CHANGELOG.md).

If you want to help with testing, here's a [org.openhab.io.openhabcloud-2.5.0-SNAPSHOT.jar](https://github.com/wborn/openhab2-addons/releases/download/openhabcloud-improvements-release/org.openhab.io.openhabcloud-2.5.0-SNAPSHOT.jar).

Do you know of any incompatibility issues with the cloud introduced by these upgrades @digitaldan, @marziman?

### CODE REVIEW

File: addons/io/org.openhab.io.openhabcloud/about.html [hunk 1]
```diff
@@ -24,5 +24,41 @@ <h3>License</h3>
 indicated below, the terms and conditions of the EPL still apply to any source code in the Content
 and such source code may be obtained at <a href="http://www.openhab.org/">openhab.org</a>.</p>
 
+<h3>Third Party Content</h3>
```
Comment (cweitkamp) [inline addons/io/org.openhab.io.openhabcloud/about.html]: :+1:

**Comment by cweitkamp** [comment]
@wborn It seems to work but I currently have the slight feeling, that I do not see updates on https://myopenhab.org/ in my Events anymore. The last entry occurred before I changed to your new version. Can you verify that?

Exposing items ans controlling them via Google Assistant works fine.

I cannot test Notifications. I have not used them before.

**Comment by wborn** [comment]
Thanks for helping with testing and reviewing @cweitkamp!

> The last entry occurred before I changed to your new version. Can you verify that?

I see the same but it also doesn't update whenever I use the 2.4.0 bundle again. So it is probably some cloud issue.

> Exposing items ans controlling them via Google Assistant works fine.

That's the most important cloud feature I use and it keeps working for me too! 

Proxying to you own instance via the cloud (https://home.myopenhab.org/start/index) also still works.

> I cannot test Notifications. I have not used them before.

I haven't used them before but they still work with the changes. :smile:

**Comment by cweitkamp** [comment]
> but it also doesn't update whenever I use the 2.4.0 bundle again.

Hm, I cannot confirm that. After switching back to 2.4.0 bundle my event log is full with recent changes.

> Proxying to your own instance via the cloud (https://home.myopenhab.org/start/index) also still works.

Yes, confirmed.

> I haven't used them before but they still work with the changes. smile

I will setup some `sendLogNotification()` in my rules to get in touch with it.

**Comment by wborn** [comment]
Today I again got disconnected from the Cloud with all updated dependencies in place. :disappointed: 

It looks like the Cloud is not properly keeping track of offline/online instances @digitaldan . 

According to the `CloudClient` logging the connection should still be up after several disconnects:

```
2019-01-05 19:38:59.570 [DEBUG] [io.openhabcloud.internal.CloudClient] - Socket.IO disconnected
2019-01-05 19:38:59.570 [INFO ] [io.openhabcloud.internal.CloudClient] - Disconnected from the openHAB Cloud service (UUID = xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx, base URL = http://localhost:8080)
2019-01-05 19:39:03.892 [ERROR] [io.openhabcloud.internal.CloudClient] - Socket.IO error: {}
io.socket.engineio.client.EngineIOException: xhr poll error
	at io.socket.engineio.client.Transport.onError(Transport.java:63) [246:org.openhab.io.openhabcloud:2.5.0.201901022358]
	at io.socket.engineio.client.transports.PollingXHR.access$100(PollingXHR.java:26) [246:org.openhab.io.openhabcloud:2.5.0.201901022358]
	at io.socket.engineio.client.transports.PollingXHR$6$1.run(PollingXHR.java:140) [246:org.openhab.io.openhabcloud:2.5.0.201901022358]
	at io.socket.thread.EventThread$2.run(EventThread.java:80) [246:org.openhab.io.openhabcloud:2.5.0.201901022358]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
Caused by: java.io.IOException: 502
	at io.socket.engineio.client.transports.PollingXHR$Request$1.onResponse(PollingXHR.java:234) ~[?:?]
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:206) ~[?:?]
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32) ~[?:?]
	... 3 more
2019-01-05 19:39:03.897 [DEBUG] [io.openhabcloud.internal.CloudClient] - Socket.IO disconnected
2019-01-05 19:39:03.897 [INFO ] [io.openhabcloud.internal.CloudClient] - Disconnected from the openHAB Cloud service (UUID = xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx, base URL = http://localhost:8080)
2019-01-05 19:39:13.426 [ERROR] [io.openhabcloud.internal.CloudClient] - Socket.IO error: {}
io.socket.engineio.client.EngineIOException: xhr poll error
	at io.socket.engineio.client.Transport.onError(Transport.java:63) [246:org.openhab.io.openhabcloud:2.5.0.201901022358]
	at io.socket.engineio.client.transports.PollingXHR.access$100(PollingXHR.java:26) [246:org.openhab.io.openhabcloud:2.5.0.201901022358]
	at io.socket.engineio.client.transports.PollingXHR$6$1.run(PollingXHR.java:140) [246:org.openhab.io.openhabcloud:2.5.0.201901022358]
	at io.socket.thread.EventThread$2.run(EventThread.java:80) [246:org.openhab.io.openhabcloud:2.5.0.201901022358]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
Caused by: java.io.IOException: 502
	at io.socket.engineio.client.transports.PollingXHR$Request$1.onResponse(PollingXHR.java:234) ~[?:?]
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:206) ~[?:?]
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32) ~[?:?]
	... 3 more
2019-01-05 19:39:13.430 [DEBUG] [io.openhabcloud.internal.CloudClient] - Socket.IO disconnected
2019-01-05 19:39:13.431 [INFO ] [io.openhabcloud.internal.CloudClient] - Disconnected from the openHAB Cloud service (UUID = xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx, base URL = http://localhost:8080)
2019-01-05 19:39:21.398 [DEBUG] [io.openhabcloud.internal.CloudClient] - Socket.IO connected
2019-01-05 19:39:21.398 [INFO ] [io.openhabcloud.internal.CloudClient] - Connected to the openHAB Cloud service (UUID = xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx, base URL = http://localhost:8080)
```

Also if I raise logging on `io.socket` there is a connection because it ping/pongs normally:

```
2019-01-05 21:59:10.352 [DEBUG] [io.socket.engineio.client.Socket    ] - writing ping packet - expecting pong within 60000ms
2019-01-05 21:59:10.353 [DEBUG] [io.socket.engineio.client.Socket    ] - flushing 1 packets in socket
2019-01-05 21:59:10.379 [DEBUG] [io.socket.engineio.client.Socket    ] - socket received: type 'pong', data 'null'
2019-01-05 21:59:35.382 [DEBUG] [io.socket.engineio.client.Socket    ] - writing ping packet - expecting pong within 60000ms
2019-01-05 21:59:35.384 [DEBUG] [io.socket.engineio.client.Socket    ] - flushing 1 packets in socket
2019-01-05 21:59:35.418 [DEBUG] [io.socket.engineio.client.Socket    ] - socket received: type 'pong', data 'null'
```

---

The cloud keeps showing:

offline


---

As soon you bundle:restart the `openHAB Cloud Connector Bundle` the Cloud again properly registers the instance as online:

online

There are already a few community topics about this:
* [Openhab 2 losing cloud connection](https://community.openhab.org/t/openhab-2-losing-cloud-connection/36550?u=wborn)
* [Myopenhab Socket.IO error, app not working](https://community.openhab.org/t/myopenhab-socket-io-error-app-not-working/34420?u=wborn)

**Comment by andre2308** [comment]
I‘am on 2.5.0 Snapshot 1491 and get These error sometimes in log:

`2019-01-09 03:25:28.478 [ERROR] [io.openhabcloud.internal.CloudClient] - Socket.IO error: {}
io.socket.engineio.client.EngineIOException: websocket error
	at io.socket.engineio.client.Transport.onError(Transport.java:71) [235:org.openhab.io.openhabcloud:2.5.0.201901061034]
	at io.socket.engineio.client.transports.WebSocket.access$500(WebSocket.java:28) [235:org.openhab.io.openhabcloud:2.5.0.201901061034]
	at io.socket.engineio.client.transports.WebSocket$2$5.run(WebSocket.java:150) [235:org.openhab.io.openhabcloud:2.5.0.201901061034]
	at io.socket.thread.EventThread$2.run(EventThread.java:80) [235:org.openhab.io.openhabcloud:2.5.0.201901061034]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
Caused by: java.io.EOFException
	at okio.RealBufferedSource.require(RealBufferedSource.java:59) ~[?:?]
	at okio.RealBufferedSource.readByte(RealBufferedSource.java:72) ~[?:?]
	at okhttp3.internal.ws.WebSocketReader.readHeader(WebSocketReader.java:115) ~[?:?]
	at okhttp3.internal.ws.WebSocketReader.processNextFrame(WebSocketReader.java:104) ~[?:?]
	at okhttp3.internal.ws.RealWebSocket.readMessage(RealWebSocket.java:97) ~[?:?]
	at okhttp3.ws.WebSocketCall.createWebSocket(WebSocketCall.java:152) ~[?:?]
	at okhttp3.ws.WebSocketCall.access$000(WebSocketCall.java:41) ~[?:?]
	at okhttp3.ws.WebSocketCall$1.onResponse(WebSocketCall.java:97) ~[?:?]
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:126) ~[?:?]
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32) ~[?:?]
	... 3 more`


I Hope this help you.. when needed i can provide make a Debug log or something.

**Comment by LuckyMallari** [comment]
I am not able to reproduce at will also. My cloud instance is monitored and have not seen outages. It sends me an email when the connector could not connect to my OH. So far no emails

**Comment by wborn** [comment]
After >3 weeks testing I didn't run into any new issues compared to the original implementation.

**Pull request merged**